web/frontend/src/api/robot.ts

```diff
@@ -20,3 +21,43 @@ export const fetchCurrentOps = (client: Client) => {
     });
   });
 };
+
+export const getResourceNames = (client: Client) => {
+  const request = new robotApi.ResourceNamesRequest();
+
+  return new Promise<ResourceName[]>((resolve, reject) => {
+    client.robotService.resourceNames(request, new grpc.Metadata(), (error, response) => {
+      if (error) {
+        reject(error);
+        return;
+      }
+
+      if (!response) {
+        reject(new Error('An unexpected issue occured.'));
+        return;
+      }
+
+      resolve(response.toObject().resourcesList);
```

Comment:
If possible, move `response` transformations to a `then` to allow `new Promise(...)` to deal only with promisification of the callback

Reply:
Updated across the board for the `api` folder 👍